freer-extras/src/Control/Monad/Freer/Extras/Modify.hs

```diff
@@ -91,95 +59,14 @@ inserting effects at the *end* of the list is tricky.
-weakenEnd8 u = case decomp u of
-    Left u' -> weaken $ weakenEnd7 u'
-    Right t -> inj t
-
-weakenEnd9 :: forall effs a b c d e f g h i. Union '[a, b, c, d, e, f, g, h, i] ~> Union (a ': b ': c ': d ': e ': f ': g ': h ': i ': effs)
-weakenEnd9 u = case decomp u of
-    Left u' -> weaken $ weakenEnd8 u'
-    Right t -> inj t
-
-weakenEnd10 :: forall effs a b c d e f g h i j. Union '[a, b, c, d, e, f, g, h, i, j] ~> Union (a ': b ': c ': d ': e ': f ': g ': h ': i ': j ': effs)
-weakenEnd10 u = case decomp u of
-    Left u' -> weaken $ weakenEnd9 u'
-    Right t -> inj t
-
-weakenEnd11 :: forall effs a b c d e f g h i j k. Union '[a, b, c, d, e, f, g, h, i, j, k] ~> Union (a ': b ': c ': d ': e ': f ': g ': h ': i ': j ': k ': effs)
-weakenEnd11 u = case decomp u of
-    Left u' -> weaken $ weakenEnd10 u'
-    Right t -> inj t
-
-weakenEnd12 :: forall effs a b c d e f g h i j k l. Union '[a, b, c, d, e, f, g, h, i, j, k, l] ~> Union (a ': b ': c ': d ': e ': f ': g ': h ': i ': j ': k ': l ': effs)
-weakenEnd12 u = case decomp u of
-    Left u' -> weaken $ weakenEnd11 u'
-    Right t -> inj t
-
-weakenEnd13 :: forall effs a b c d e f g h i j k l m. Union '[a, b, c, d, e, f, g, h, i, j, k, l, m] ~> Union (a ': b ': c ': d ': e ': f ': g ': h ': i ': j ': k ': l ': m ': effs)
-weakenEnd13 u = case decomp u of
-    Left u' -> weaken $ weakenEnd12 u'
-    Right t -> inj t
-
-weakenEnd14 :: forall effs a b c d e f g h i j k l m n. Union '[a, b, c, d, e, f, g, h, i, j, k, l, m, n] ~> Union (a ': b ': c ': d ': e ': f ': g ': h ': i ': j ': k ': l ': m ': n ': effs)
-weakenEnd14 u = case decomp u of
-    Left u' -> weaken $ weakenEnd13 u'
-    Right t -> inj t
-
-weakenEnd15 :: forall effs a b c d e f g h i j k l m n o. Union '[a, b, c, d, e, f, g, h, i, j, k, l, m, n, o] ~> Union (a ': b ': c ': d ': e ': f ': g ': h ': i ': j ': k ': l ': m ': n ': o ': effs)
-weakenEnd15 u = case decomp u of
-    Left u' -> weaken $ weakenEnd14 u'
-    Right t -> inj t
-
-weakenEnd16 :: forall effs a b c d e f g h i j k l m n o p. Union '[a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p] ~> Union (a ': b ': c ': d ': e ': f ': g ': h ': i ': j ': k ': l ': m ': n ': o ': p ': effs)
-weakenEnd16 u = case decomp u of
-    Left u' -> weaken $ weakenEnd15 u'
-    Right t -> inj t
-
-weakenEnd17 :: forall effs a b c d e f g h i j k l m n o p q. Union '[a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q] ~> Union (a ': b ': c ': d ': e ': f ': g ': h ': i ': j ': k ': l ': m ': n ': o ': p ': q ': effs)
-weakenEnd17 u = case decomp u of
-    Left u' -> weaken $ weakenEnd16 u'
-    Right t -> inj t
-
-weakenEnd18 :: forall effs a b c d e f g h i j k l m n o p q r. Union '[a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r] ~> Union (a ': b ': c ': d ': e ': f ': g ': h ': i ': j ': k ': l ': m ': n ': o ': p ': q ': r ': effs)
-weakenEnd18 u = case decomp u of
-    Left u' -> weaken $ weakenEnd17 u'
-    Right t -> inj t
+class CanWeakenEnd as effs where
```

Comment:
This is actually a very straightforward implementation, not sure why I didn't come up with it. LGTM!

Reply:
Yeah, I started out with something much more complicated (thinking you'd need to do something like `raiseEnd @7`) and once I got something working it turned out it kept working after stripping stuff out bit by bit.